### PR TITLE
feature/483 refactor-external-data

### LINKED ISSUE #483: Refactor external_data module
This module should disappear, becoming two separate corrections:
 - astrometry correction using external data
 - flux calibration using external data
 
Auxiliary methods for dealing with the external data can be placed in a different module outside the library of corrections.

### PR DESCRIPTION
**Pull request by PabloCorcho**
This PR includes a refactoring of the external data module. Fixes #483

### CODE REVIEW

**Comment by PabloCorcho** [comment]
@paranoya @miguelglezb I think it is safer to start merging just the refactoring of the astrometry, and leave for a next PR the flux calibration using the external data (which might require additional features that do not exist at the moment).  Please have a look at the changes, and let me know if you have any comments.

File: src/pykoala/data_container.py [hunk 1]
```diff
@@ -1096,8 +1116,8 @@ def from_fits(cls, path, hdul_extension_map=None, **kwargs):
         - cube: Cube
             An instance of a `pykoala.cubing.Cube`.
         """
-        with fits.open(path) as hdul:
-            return cls(hdul, hdul_extension_map=hdul_extension_map, **kwargs)
+        hdul = fits.open(path)
```
Comment (miguelglezb) [inline src/pykoala/data_container.py:1119]: I am curious if this change affects the memory storage for a large set of data

File: tutorials/compare_with_external_photo.ipynb
Comment (miguelglezb) [inline tutorials/compare_with_external_photo.ipynb:1]: Is there an actual change in the notebook or just the rerun of it after implementing the changes? If so, I recommend to omit adding it unless we actually change an input cell.

**Review by miguelglezb — APPROVED** [review]
Except for some comments, I'm okay with this PR.

File: tutorials/compare_with_external_photo.ipynb
Comment (PabloCorcho) [reply tutorials/compare_with_external_photo.ipynb:1]: There are indeed a couple of changes, but I also wanted to push a clean version of the tutorials without the results on every cell (the users should get that by themselves without spoilers). I should get used to clean the notebook before adding changes.

File: src/pykoala/data_container.py (hunk 1 above)
Comment (PabloCorcho) [reply src/pykoala/data_container.py:1119]: In principle no, the data is not loaded into memory unless explicitly stated.

**Pull request merged**